spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkRInterpreter.java

```diff
@@ -59,12 +59,18 @@ public void open() throws InterpreterException {
     String sparkRLibPath;
 
     if (System.getenv("SPARK_HOME") != null) {
+      // local or yarn-client mode when SPARK_HOME is specified
       sparkRLibPath = System.getenv("SPARK_HOME") + "/R/lib";
-    } else {
+    } else if (System.getenv("ZEPPELIN_HOME") != null){
+      // embedded mode when SPARK_HOME is not specified
       sparkRLibPath = System.getenv("ZEPPELIN_HOME") + "/interpreter/spark/R/lib";
       // workaround to make sparkr work without SPARK_HOME
       System.setProperty("spark.test.home", System.getenv("ZEPPELIN_HOME") + "/interpreter/spark");
+    } else {
+      // yarn-cluster mode
+      sparkRLibPath = "sparkr";
```

Comment:
hmm, in yarn cluster mode SPARK_HOME is not set?

Reply:
There's no SPARK_HOME in driver of yarn cluster mode. As the driver run in yarn container.